test/Test/Database/LSMTree/ModelIO/Class.hs

```diff
@@ -0,0 +1,87 @@
+{-# LANGUAGE KindSignatures #-}
+{-# LANGUAGE LambdaCase     #-}
+{-# LANGUAGE TypeFamilies   #-}
+module Test.Database.LSMTree.ModelIO.Class (
+    LookupResult (..)
+  , TableHandle (..)
+  ) where
+
+import           Data.Bifunctor (second)
+import           Data.Kind (Constraint, Type)
+import           Database.LSMTree.Common (IOLike)
+import           Database.LSMTree.Model.Normal (Update (..))
+import qualified Database.LSMTree.ModelIO.Normal as M
+import qualified Database.LSMTree.Normal as R
+
+-- | Result of a single point lookup.
+--
+-- TODO: should this be shared?
+data LookupResult k v blobref =
+    NotFound      !k
+  | Found         !k !v
+  | FoundWithBlob !k !v !blobref
+  deriving (Eq, Show)
+
+class TableHandle h where
+    type Session h :: (Type -> Type) -> Type
+    type TableConfig h :: Type
+    type BlobRef h :: Type -> Type
+
+    -- TODO: should there be only one definition of SomeSerialisationConstraint in the library?
```

Comment:
I would lean towards having separate constraints, but maybe @dcoutts can chime in

Reply:
What would be different in model in real constraints? What *benefit* would it give?

Reply:
I can't immediately think of any reason the serialisation constraints would need to be different for the normal and monoidal APIs. We might plausibly want something _extra_ for the monodial case, but that can be a subclass or extra class that doesn't need to be shared. So yes I'd say share the serialisation class.

Reply:
The discussion was about sharing the class between model and real implementation, not between normal and monoidal tables.

IMO I'd have just one class. But please decide about the other axis too.